tbmalt/io/dataset.py

```diff
@@ -1,66 +1,109 @@
+    Arguments:
+        dataset: An dictionary which stores all the data.
+        labels: A list of label for each sample.
+        targets: Names of machine learned targets.
+        seed: Seed to generate random numbers, this will be used to
+            generate indices when loading samples.
 
-    @classmethod
-    def load_reference(cls, dataset: str, size: int, properties: List):
-        """Load reference from h5py type data."""
-        data = {pro: [] for pro in properties}
-        positions, numbers = [], []
+    """
 
-        with h5py.File(dataset, 'r') as f:
-            gg = f['global_group']
-            molecule_specie = gg.attrs['molecule_specie_global']
+    def __init__(self, dataset: Dict, labels: List[str], targets, seed):
+        self.targets = targets
+        self.dataset = dataset
+        self.labels = labels
 
-            size_mol = int(size / len(molecule_specie))
+        size = len(dataset['atomic_numbers'])
+        random.seed(seed)
+        self.random_idx = random.sample(range(size), size)
 
-            # add atom name and atom number
-            for imol_spe in molecule_specie:
-                g = f[imol_spe]
-                g_size = g.attrs['n_molecule']
-                size_mol = min(g_size, size_mol)
+    def __getitem__(self, idx: Tensor) -> Dict:
+        """Get data values according to input indices."""
+        atomic_numbers = self.dataset['atomic_numbers']
+        positions = self.dataset['positions']
 
-                ind = random.sample(torch.arange(g_size).tolist(), size_mol)
+        # Select different properties
+        data = {target: pack([self.dataset[target][ii] for ii in idx])
+                for target in self.targets}
 
-                # loop for the same molecule specie
-                for imol in ind:
-                    for ipro in properties:  # loop for each property
-                        idata = g[str(imol + 1) + ipro][()]
-                        data[ipro].append(torch.from_numpy(idata))
+        # Select data with index
+        data['atomic_numbers'] = pack([atomic_numbers[ii] for ii in idx])
+        data['positions'] = pack([positions[ii] for ii in idx])
 
-                    positions.append(torch.from_numpy(g[str(imol + 1) + 'position'][()]))
-                    numbers.append(torch.from_numpy(g.attrs['numbers']))
+        return data
 
-        return cls(numbers, positions, data)
+    @classmethod
+    def load_reference(cls, groups: Union[Group, List[Group]],
+                       targets: List[str],
+                       sizes: Union[List[int], int] = None,
+                       seed: int = 1):
+        """Load reference from h5py type data.
+
+        Arguments:
+            groups: The groups in h5py binary files. The input groups
+                can be a single group or a list of groups.
+            targets: Loading targets, such as dipole, charge, etc.
+            sizes: Loading size of each group.
```

Comment:
Is `sizes` the size of each batch. If so we should probably call this `batch_sizes` just to make it clear what it is doing.